### PR TITLE
Add telemetry events for webworker

### PR DESCRIPTION
**Pull request by ritikramuka**
This pull request primarily focuses on enhancing telemetry in the web client. It introduces new telemetry events, modifies error handling, and adds telemetry messages to several functions in the `webworker.js` and `extension.ts` files.

Here are the key changes:

Telemetry Enhancements:

* [`src/web/client/common/constants.ts`](diffhunk://#diff-8769a22aef8333b09b6f98fe24f9c95ece22948a556d3d9f64528f4a6642add3R76-R77): Added `SEND_INFO_TELEMETRY` and `SEND_ERROR_TELEMETRY` to the `workerEventMessages` enum to support new telemetry events.

* [`src/web/client/common/worker/webworker.js`](diffhunk://#diff-8e0adceeebffaef7b18326a30ca124d6e2d8448e6732876fdf21ee42b14897c1R85-R89): Added telemetry messages at the start of the `loadContainer` function, when a member is removed, and when an error occurs. [[1]](diffhunk://#diff-8e0adceeebffaef7b18326a30ca124d6e2d8448e6732876fdf21ee42b14897c1R85-R89) [[2]](diffhunk://#diff-8e0adceeebffaef7b18326a30ca124d6e2d8448e6732876fdf21ee42b14897c1R112-R122) [[3]](diffhunk://#diff-8e0adceeebffaef7b18326a30ca124d6e2d8448e6732876fdf21ee42b14897c1R165-R187)

* [`src/web/client/extension.ts`](diffhunk://#diff-1429bea43439e067ae9056ff3f9f71203533f7d3ca168f0d5469aa1b9a6e0360R390-R403): Added handling for `SEND_INFO_TELEMETRY` and `SEND_ERROR_TELEMETRY` events in the `createWebWorkerInstance` function.

Error Handling Changes:

* [`src/web/client/common/worker/webworker.js`](diffhunk://#diff-8e0adceeebffaef7b18326a30ca124d6e2d8448e6732876fdf21ee42b14897c1L119-L124): Changed the `loadContainer` function to throw an error instead of returning null when the `getUserIdByConnectionId` fails.

### CODE REVIEW

File: src/web/client/extension.ts [hunk 1]
```diff
@@ -380,6 +380,12 @@ export function createWebWorkerInstance(
                             WebExtensionContext.userCollaborationProvider.refresh();
                             WebExtensionContext.quickPickProvider.refresh();
                         }
+                        if (data.type === Constants.workerEventMessages.TELEMETRY) {
+                            WebExtensionContext.telemetry.sendInfoTelemetry(
```
Comment (tyaginidhi) [inline src/web/client/extension.ts:384]: you would want to log info and error as two different events

File: src/web/client/common/worker/webworker.js [hunk 2]
```diff
@@ -146,10 +162,25 @@ async function loadContainer(config, swpId, entityInfo) {
                     containerId: swpId,
                     entityId: userEntityIdArray,
                 });
+
+                await self.postMessage({
+                    type: "telemetry",
+                    eventName: "client-data success",
```
Comment (tyaginidhi) [inline src/web/client/common/worker/webworker.js:168]: is this event name aligning with current naming convention? Refer current event constants.

File: src/web/client/extension.ts (hunk 1 above)
Comment (ritikramuka) [reply src/web/client/extension.ts:384]: addressed!

File: src/web/client/common/worker/webworker.js (hunk 2 above)
Comment (ritikramuka) [reply src/web/client/common/worker/webworker.js]: addressed

**Review by tyaginidhi — APPROVED** [review]

File: src/web/client/common/worker/webworker.js [hunk 3]
```diff
@@ -82,6 +82,11 @@ let intital = false;
 
 async function loadContainer(config, swpId, entityInfo) {
     try {
+        self.postMessage({
```
Comment (BidishaMS) [inline src/web/client/common/worker/webworker.js:85]: Are they using webextensiontelemetry ? If yes then we should be good. If not then what is the data residency part for this scenarios

Comment (ritikramuka) [reply src/web/client/common/worker/webworker.js:85]: yes they uses webExtensionTelemetry [link](https://github.com/microsoft/powerplatform-vscode/pull/824/files#:~:text=if%20(data.type%20%3D%3D%3D%20Constants.workerEventMessages.SEND_INFO_TELEMETRY)%20%7B)

**Review by BidishaMS — APPROVED** [review]

**Pull request merged**